### PR TITLE
feat: impl to_json

### PR DESCRIPTION
**Pull request by yubing744**
## Summary

Summary about this PR

- Closes #issue
https://github.com/rooch-network/rooch/issues/2655

### CODE REVIEW

**Comment by yubing744** [comment]
@jolestar help review

File: frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs [hunk 1]
```diff
@@ -1,36 +1,48 @@
...
+use move_core_types::gas_algebra::{InternalGas, InternalGasPerByte, NumBytes};
+use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::StructTag;
 use move_core_types::language_storage::TypeTag;
 use move_core_types::u256::U256;
-use move_core_types::value::MoveStructLayout;
+use move_core_types::value::MoveStruct;
+use move_core_types::value::MoveValue as CoreMoveValue;
+use move_core_types::value::{MoveFieldLayout, MoveStructLayout, MoveTypeLayout};
 use move_core_types::vm_status::StatusCode;
```
Comment (jolestar) [inline frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs:25]: Why rename MoveValue to CoreMoveValue and Value to MoveValue? It's easy to get confused.

File: frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs [hunk 2]
```diff
@@ -275,34 +290,358 @@ fn native_from_json(
...
-    )];
+    let natives = [
+        (
+            "native_from_json",
+            make_native(gas_params.from_bytes, native_from_json),
+        ),
+        (
+            "native_to_json",
+            make_native(gas_params.to_bytes, native_to_json),
+        ),
```
Comment (jolestar) [inline frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs:643]: The new native function gas_params needs to be optional, like https://github.com/rooch-network/rooch/blob/a3ae131ea83a9da6bce3d8461e2a95860f581f5c/frameworks/rooch-framework/src/natives/rooch_framework/bitcoin_address.rs#L247-L255

**Comment by jolestar** [comment]
cc @steelgeek091

File: frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs (hunk 2 above)
Comment (yubing744) [reply frameworks/moveos-stdlib/src/natives/moveos_stdlib/json.rs:643]: OK, I'll fix it.

**Comment by jolestar** [comment]
Find a native gas param bug. The gas config of json needs to be corrected. Create a thread to discuss: https://discord.com/channels/1078938449974935592/1286723707292880896

https://github.com/rooch-network/rooch/blob/a3ae131ea83a9da6bce3d8461e2a95860f581f5c/frameworks/rooch-framework/src/natives/gas_parameter/json.rs#L4-L11

**Comment by yubing744** [comment]
> Find a native gas param bug. The gas config of json needs to be corrected. Create a thread to discuss: https://discord.com/channels/1078938449974935592/1286723707292880896
> 
> https://github.com/rooch-network/rooch/blob/a3ae131ea83a9da6bce3d8461e2a95860f581f5c/frameworks/rooch-framework/src/natives/gas_parameter/json.rs#L4-L11

@jolestar Added missing GAS initial configuration

**Review by jolestar — APPROVED** [review]

**Pull request merged**